openshift-kube-apiserver/admission/autoscaling/managementcpusoverride/admission.go

```
@@ -588,7 +591,11 @@ func (a *managementCPUsOverride) Validate(ctx context.Context, attr admission.At
 		}
 
 		for resourceName, c := range containersWorkloadResources {
-			allErrs = append(allErrs, field.Invalid(field.NewPath("spec.containers.resources.requests"), c.Resources.Requests, fmt.Sprintf("the pod without workload annotations can not have containers with workload resources %q", resourceName)))
+			if isDebugPod(pod.Annotations) {
+				warning.AddWarning(ctx, "", "You must pass --keep-annotations parameter to the debug command when trying to debug a pod with workload partitioning resources.")
+			} else {
```

why the `else`? Shouldn't we return the error?